pkg/activator/handler/handler_test.go

```diff
@@ -12,11 +12,13 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package handler
+package handler_test
```

Comment:
this change should be unnecessary.

Reply:
Also this would remove the necessity of public fields.

Reply:
It is good practice to avoid accessing private fields.
The overuse of private fields in tests is what made a seemingly simple change much harder.
Required fields with no default value are passed through the constructor.
Fields with a default value are public.

Reply:
There is no reason to have public fields on this type altogether. Nor should the type be public.
Go doesn't do OOP. This is a prime example of client side defined interface (ServeHTTP is the only thing that the clients have to know about the type).

Reply:
Ok. It's reverted.